worlds/oot_soh/LogicHelpers.py

```diff
@@ -701,4 +705,103 @@ def can_detonate_upright_bomb_flower(state: CollectionState, world: "SohWorld")
+            (world.options.rainbow_bridge.value == 2 and (stone_count(state, world) >= world.options.rainbow_bridge_stones_required.value)) or
+            (world.options.rainbow_bridge.value == 3 and (medallion_count(state, world) >= world.options.rainbow_bridge_medallions_required.value)) or
+            (world.options.rainbow_bridge.value == 4 and (stone_count(state, world) + medallion_count(state, world) >= world.options.rainbow_bridge_dungeon_rewards_required.value)) or
+            (world.options.rainbow_bridge.value == 5 and (dungeon_count() >= world.options.rainbow_bridge_dungeons_required.value)) or
+            (world.options.rainbow_bridge.value == 6 and (get_gs_count(state, world) >= world.options.rainbow_bridge_skull_tokens_required.value))
+            (world.options.rainbow_bridge.value == 7 and (has_item(Items.GREG_THE_GREEN_RUPEE) and bool(world.options.rainbow_bridge_greg_reward.value))))
+
+# This version includes Greg Wildcard
+# def can_build_rainbow_bridge(state: CollectionState, world: "SohWorld") -> bool:
+#     return (world.options.rainbow_bridge.value == 1 or
+#             (world.options.rainbow_bridge.value == 0 and has_item(Items.SHADOW_MEDALLION, state, world) and has_item(Items.SPIRIT_MEDALLION, state, world) and can_use(Items.LIGHT_ARROW, state, world)) or
+#             (world.options.rainbow_bridge.value == 2 and (stone_count(state, world) + (has_item(Items.GREG_THE_GREEN_RUPEE) and bool(world.options.rainbow_bridge_greg_reward.value)) >= world.options.rainbow_bridge_stones_required.value)) or
+#             (world.options.rainbow_bridge.value == 3 and (medallion_count(state, world) + (has_item(Items.GREG_THE_GREEN_RUPEE) and bool(world.options.rainbow_bridge_greg_reward.value)) >= world.options.rainbow_bridge_medallions_required.value)) or
+#             (world.options.rainbow_bridge.value == 4 and (stone_count(state, world) + medallion_count(state, world) + (has_item(Items.GREG_THE_GREEN_RUPEE) and bool(world.options.rainbow_bridge_greg_reward.value)) >= world.options.rainbow_bridge_dungeon_rewards_required.value)) or
+#             (world.options.rainbow_bridge.value == 5 and (dungeon_count() + (has_item(Items.GREG_THE_GREEN_RUPEE) and bool(world.options.rainbow_bridge_greg_reward.value)) >= world.options.rainbow_bridge_dungeons_required.value)) or
+#             (world.options.rainbow_bridge.value == 6 and (get_gs_count(state, world) >= world.options.rainbow_bridge_skull_tokens_required.value))
+#             (world.options.rainbow_bridge.value == 7 and (has_item(Items.GREG_THE_GREEN_RUPEE) and bool(world.options.rainbow_bridge_greg_reward.value))))
+
+def get_gs_count(state: CollectionState, world: "SohWorld") -> int:
+    return state.count(Items.GOLD_SKULLTULA_TOKEN, world.player)
+
+# We don't have any options related to this yet
+# def can_trigger_lacs(state: CollectionState, world: "SohWorld") -> bool:
+#     return ()
+
```

Comment:
How so? This is what holds ganon's boss key and is dependent on what that is set to, defaulting to spirit and shadow medallion if GBK is shuffled anywhere. I don't need it to accept the PR, but just wanted to know for sure if this is what you meant.